commit febefc42c2e4c0149f03bbb9bf05a7882e5fc037
Author: verevka <verevka86@gmail.com>
Date:   Sat Feb 25 20:22:58 2017 +0300

    fix MT6797
    
    Change-Id: I068aedb38287d5a12b7d5a2bae98363dc5ad6554

diff --git a/init/init.cpp b/init/init.cpp
index 6e4fd9c..2563438 100755
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -366,7 +366,7 @@ static void export_kernel_boot_props() {
         { "ro.boot.mode",       "ro.bootmode",   "unknown", },
         { "ro.boot.baseband",   "ro.baseband",   "unknown", },
         { "ro.boot.bootloader", "ro.bootloader", "unknown", },
-        { "ro.boot.hardware",   "ro.hardware",   "mt6752", },
+        { "ro.boot.hardware",   "ro.hardware",   "mt6797", },
 #ifndef IGNORE_RO_BOOT_REVISION
         { "ro.boot.revision",   "ro.revision",   "0", },
 #endif
